zuul-core/src/main/java/com/netflix/zuul/context/Debug.java

```diff
@@ -121,6 +121,10 @@ public static void addRequestDebug(SessionContext ctx, String line) {
      */
     public static void compareContextState(String filterName, SessionContext context, SessionContext copy) {
         // TODO - only comparing Attributes. Need to compare the messages too.
+
+        // Ensure that the routingDebug property already exists, otherwise we'll have a ConcurrentModificationException below
+        getRoutingDebug(context);
```

Comment:
Perhaps we should add this line too:
```
getRequestDebug(ctx);
```

Reply:
Not convinced it's necessary, as this code doesn't do anything with the request debug list.  Overall this code should be refactored to avoid mutability and getter side-effects where possible; this is just a simple/quick fix for the issue to stop the bleeding.

Reply:
Sounds good to me.